### DISCUSSION TITLE
How to change default database connection while testing ?

### DISCUSSION THREAD

**Discussion by lepresk** [adonisjs/core#4493] [Help] [answered]
Hello everyone,

I'm currently facing an issue regarding changing the default database connection to `sqliteTest` or something else in the testing environment. Does anyone have any suggestions or tips they could share?

Here's a snippet of my code:

```
import app from '@adonisjs/core/services/app'
import { defineConfig } from '@adonisjs/lucid'

const dbConfig = defineConfig({
  connection: 'sqlite',
  connections: {
    sqlite: {
      client: 'better-sqlite3',
      connection: {
        filename: app.tmpPath('db.sqlite3'),
      },
      useNullAsDefault: true,
      migrations: {
        naturalSort: true,
        paths: ['database/migrations'],
      },
    },
    sqliteTest: {
      client: 'better-sqlite3',
      connection: {
        filename: app.tmpPath('test.sqlite3'),
      },
      useNullAsDefault: true,
      migrations: {
        naturalSort: true,
        paths: ['database/migrations'],
      },
    },
  },
})

export default dbConfig
```

Thank you in advance !

**Comment by Julien-R44** [comment, marked as answer]
- Define a `DB_CONNECTION=sqlite` variable in your environment variables ( `.env` + `env.ts` )
- Define a `DB_CONNECTION=sqliteTest` variable in your `.env.test` file
- Use this `DB_CONNECTION` in your `config/database.ts` file: 

```ts
const dbConfig = defineConfig({
  connection: env.get('DB_CONNECTION'), // 👈👈👈👈
  connections: {
    sqlite: {
     // ...
    },
    sqliteTest: {
      // ...
     }
  },
})

export default dbConfig
```

And when you run your tests, the `.env.test` will be used, so your `sqliteTest` connection defined in your config will be used

see https://docs.adonisjs.com/guides/testing#environment-variables

**Comment by lepresk** [comment]
Thank you so much for your quick response!
